src/app/qgsdockablewidgethelper.h

```diff
@@ -0,0 +1,77 @@
+/***************************************************************************
+  qgsdockablewidgethelper.h
+  --------------------------------------
+  Date                 : January 2022
+  Copyright            : (C) 2022 by Belgacem Nedjima
+  Email                : belgacem dot nedjima at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSDOCKABLEWIDGETHELPER_H
+#define QGSDOCKABLEWIDGETHELPER_H
+
+#include "qgis_app.h"
+#include "qgsdockwidget.h"
```

Comment:
this could be just a forward declaration `class QgsDockWidget;` and #include in cpp file?